Sources/HTTP/SimpleSocket/SimpleServerSocket.swift

```diff
@@ -0,0 +1,152 @@
+// This source file is part of the Swift.org Server APIs open source project
+//
+// Copyright (c) 2017 Swift Server API project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+//
+
+import Foundation
+import Dispatch
+
+internal class SimpleServerSocket {
+    internal var socketfd: Int32 = -1
+    
+    internal var listeningPort: Int32 = -1
+    
+    internal private(set) var isListening = false
+    
+    internal private(set) var isConnected = false
+
+    internal func socketRead(into readBuffer: inout UnsafeMutablePointer<Int8>, maxLength:Int) throws -> Int {
```

Comment:
Why not use `Data`?

Reply:
See #37 - This makes it possible to measure Data vs DispatchData, which was part of the goal of this exercise.

Reply:
OK so it's a temporary thing - cool.